### ISSUE TITLE
gradle 6.6.1 compile DockerMock.groovy unsupported Target MODULE

### ISSUE THREAD

**Issue by stchar** [jenkinsci/JenkinsPipelineUnit#292]
@nre-ableton

Nik, just found an issue  with gralde `comileGroovy:` step. Have ever you faced with it?

```bash
$ gradle test

Welcome to Gradle 6.6.1!

Here are the highlights of this release:
 - Experimental build configuration caching
 - Built-in conventions for handling credentials
 - Java compilation supports --release flag

For more details see https://docs.gradle.org/6.6.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)
> Task :compileGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGroovy'.
> BUG! exception in phase 'class generation' in source unit '.../JenkinsPipelineUnit/src/main/groovy/com/lesfurets/jenkins/unit/DockerMock.groovy' unsupported Target MODULE

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

```

**Comment by nre-ableton** [comment]
I get a different error, at least when using the gradle wrapper:

```
master!JenkinsPipelineUnit$ ./gradlew test
> Task :buildSrc:compileGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildSrc:compileGroovy'.
> BUG! exception in phase 'semantic analysis' in source unit '/home/nre/Code/nre-ableton/JenkinsPipelineUnit/buildSrc/src/main/groovy/com/lesfurets/gradle/CopyDependenciesToMavenTreeTask.groovy' Unsupported class file major version 57

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```

It seems that the gradle wrapper is 5.6.3, which is quite a bit older than your version. Does the provided wrapper work for you?

We should update the wrapper at some point, I guess.

**Comment by nre-ableton** [comment]
(I guess the above error means that my JDK is not compatible with the one used to build that version of Gradle, see https://github.com/exercism/support/issues/73#issuecomment-538303598).

**Comment by nre-ableton** [comment]
@stchar So, I managed to install Gradle 6.6.1 manually, and I could then reproduce your issue locally. It's caused stupidly enough by using a newer JDK version (in my case, JDK-13). If you downgrade to JDK-8, it'll be fine.

```
~/.../nre-ableton/JenkinsPipelineUnit$ gradle --version

Welcome to Gradle 6.6.1!

Here are the highlights of this release:
 - Experimental build configuration caching
 - Built-in conventions for handling credentials
 - Java compilation supports --release flag

For more details see https://docs.gradle.org/6.6.1/release-notes.html


------------------------------------------------------------
Gradle 6.6.1
------------------------------------------------------------

Build time:   2020-08-25 16:29:12 UTC
Revision:     f2d1fb54a951d8b11d25748e4711bec8d128d7e3

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          13.0.3 (Private Build 13.0.3+3-Ubuntu-1ubuntu2)
OS:           Linux 5.4.0-7642-generic amd64

~/.../nre-ableton/JenkinsPipelineUnit$ gradle test
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :compileGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGroovy'.
> BUG! exception in phase 'class generation' in source unit '/home/nre/Code/nre-ableton/JenkinsPipelineUnit/src/main/groovy/com/lesfurets/jenkins/unit/DockerMock.groovy' unsupported Target MODULE

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 37s
1 actionable task: 1 executed
```

But:

```
~/.../nre-ableton/JenkinsPipelineUnit$ sudo update-alternatives --config java
There are 2 choices for the alternative java (providing /usr/bin/java).

  Selection    Path                                            Priority   Status
------------------------------------------------------------
* 0            /usr/lib/jvm/java-13-openjdk-amd64/bin/java      1311      auto mode
  1            /usr/lib/jvm/java-13-openjdk-amd64/bin/java      1311      manual mode
  2            /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java   1081      manual mode

Press <enter> to keep the current choice[*], or type selection number: 2
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java to provide /usr/bin/java (java) in manual mode
~/.../nre-ableton/JenkinsPipelineUnit$ java -version
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-8u265-b01-0ubuntu2~20.04-b01)
OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)
~/.../nre-ableton/JenkinsPipelineUnit$ gradle --version

------------------------------------------------------------
Gradle 6.6.1
------------------------------------------------------------

Build time:   2020-08-25 16:29:12 UTC
Revision:     f2d1fb54a951d8b11d25748e4711bec8d128d7e3

Kotlin:       1.3.72
Groovy:       2.5.12
Ant:          Apache Ant(TM) version 1.10.8 compiled on May 10 2020
JVM:          1.8.0_265 (Private Build 25.265-b01)
OS:           Linux 5.4.0-7642-generic amd64

~/.../nre-ableton/JenkinsPipelineUnit$ gradle test
(Lots of output...)
BUILD SUCCESSFUL in 18s
6 actionable tasks: 4 executed, 2 up-to-date
```

**Comment by nre-ableton** [comment]
Also, regarding updating of the Gradle wrapper, see: https://github.com/jenkinsci/JenkinsPipelineUnit/pull/293

**Comment by rtayek** [comment]
i am getting this also on windows 10.  jdk 1.15 gradle 7.0

**Comment by nre-ableton** [comment]
@dblock sure, I'll give it a look.

**Comment by nre-ableton** [comment]
Ok, I had a look at this issue again with both JDK 16/17 (using Gradle 7.3), and can't get past the `unsupported Target MODULE`. Error. everywhere you Google, the advice is to downgrade to JDK 1.8. I have no idea why this is happening, but we can leave this issue open in the backlog. Until then, I advise to go with the workaround of using JDK 1.8.

**Comment by dblock** [comment]
This issue was fixed in Groovy 2.4.12, upgrading in #426.
